File: front/src/services/statusService.ts
import api from './api';
import type { Status } from '../types';

export const statusService = {
    listar: async (): Promise<Status[]> => {
        const res = await api.get<Status[]>('/status');
        return res.data;
    },

    buscarPorId: async (id: number): Promise<Status> => {
        const res = await api.get<Status>(`/status/${id}`);
        return res.data;
    },

    criar: async (dto: Omit<Status, 'id'>): Promise<Status> => {
        const res = await api.post<Status>('/status', dto);
        return res.data;
    },

    atualizar: async (id: number, dto: Omit<Status, 'id'>): Promise<Status> => {
        const res = await api.put<Status>(`/status/${id}`, dto);
        return res.data;
    },

    deletar: async (id: number): Promise<void> => {
        await api.delete(`/status/${id}`);
    },
};

File: back/src/main/java/com/sustentacao/service/StatusService.java
package com.sustentacao.service;

import com.sustentacao.dto.StatusDTO;
import com.sustentacao.entity.Status;
import com.sustentacao.repository.StatusRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
public class StatusService {

    private final StatusRepository repository;

    public List<StatusDTO> listar() {
        return repository.findAll().stream().map(this::toDTO).collect(Collectors.toList());
    }

    public StatusDTO buscarPorId(Long id) {
        return repository.findById(id)
                .map(this::toDTO)
                .orElseThrow(() -> new RuntimeException("Status não encontrado: " + id));
    }

    public StatusDTO salvar(StatusDTO dto) {
        Status status = new Status();
        status.setDescricao(dto.getDescricao());
        return toDTO(repository.save(status));
    }

    public StatusDTO atualizar(Long id, StatusDTO dto) {
        Status status = repository.findById(id)
                .orElseThrow(() -> new RuntimeException("Status não encontrado: " + id));
        status.setDescricao(dto.getDescricao());
        return toDTO(repository.save(status));
    }

    public void deletar(Long id) {
        repository.deleteById(id);
    }

    public StatusDTO toDTO(Status status) {
        StatusDTO dto = new StatusDTO();
        dto.setId(status.getId());
        dto.setDescricao(status.getDescricao());
        return dto;
    }
}
